libs/header/p2p/exchange.go

```diff
@@ -235,18 +236,24 @@ func (ex *Exchange[H]) performRequest(
 		return nil, fmt.Errorf("no trusted peers")
 	}
 	var reqErr error
-	for i := 0; i < len(ex.trustedPeers); i++ {
-		//nolint:gosec // G404: Use of weak random number generator
-		idx := rand.Intn(len(ex.trustedPeers))
+	peers := ex.trustedPeers
```

Comment:
This is racy because slices are copied by pointer not by value. Shuffling peers shuffles them everywhere, while there is no synchronisation around Head request. We actually should allow only one Head request at a time.